docs/cmd/kn_completion.md

```diff
@@ -25,6 +25,7 @@ kn completion [SHELL] [flags]
 
  # Generate completion code for zsh
  source <(kn completion zsh)
+ compdef _kn kn
```

Comment:
this file is auto-generated

append this line after

https://github.com/knative/client/blob/0df54bc500ad960568e5d00285d3f112b9db665e/pkg/kn/commands/completion/completion.go#L39

and run
```
./hack/build.sh
```
to auto-generate the docs